code/src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -64,25 +67,40 @@ public ShooterSubsystem(ILogger logger, final RobotConfig config) {
         pidOnTargetTicks = config.Shooter.pidOnTargetTicks;
         shooterRPM = config.Shooter.shooterRPM;
 
+        m_targetRPM = 0;
+
         m_kicker = new WPI_VictorSPX(config.Shooter.feederPort);
         m_wheels = new WPI_VictorSPX(config.Shooter.spinnerPort);
         m_wheels.setInverted(true);
 
         m_feeder = new SpeedControllerGroup(m_kicker, m_wheels);
 
+        m_limelight = limelight;
+
     }
 
     /*
+     * @param rpm if -1, goes into limelight auto
      */
+    @Override
+    public void turnOnShooter(double targetRPM) {
+        if (m_shooterPID.getSetpoint() != targetRPM) {
+            return;
+        }
+        m_shooterPID.reset();
+
+        m_shooterPID.setSetpoint(targetRPM);
```

Comment:
This isn't needed right, since we set the setpoint every time in the periodic method?

Reply:
correct